### ISSUE TITLE
Reviews

### ISSUE THREAD

**Issue by ismailbhinder** [UBC-MDS/DSCI522-2425-group31_age-group-prediction#78]
- [x] Your instruction was not clear on how to run the analysis, it ended up after pasting the Jupyter lab URL on the browser. Then, what's next? Where should one find the analysis notebook? And, how to reproduce the analysis? #93 
- [x] Your 'altair-all' package version is labelled as 'altair-all=5.4.*', this mean to take any patch version (you will learn about this later semantic versioning) that solves with the current environment. Patches are updated more frequently, hence you could end up with different version of this package for each pipeline run
<https://github.com/UBC-MDS/DSCI522-2425-group31_age-group-prediction/commit/a26f78f5e3171b8312f8fdc5c5493aba52a61b0c>
in this commit we specify altair-all=5.4.1 and add the pytest package for unit test

Improve our report based on these comments.

**Comment by DongchunChen** [comment]
- [x] Please add a link to your report in the README file, I think this would make it more easily accessible. (I found it through the link posted in the initial issue comment above).
<https://github.com/UBC-MDS/DSCI522-2425-group31_age-group-prediction/commit/41f788d869626636a97c4ed224a54b73bd00bf4d>

- [x] When I was reviewing your scripts (which are otherwise well done) I noticed that the model evaluation script re-creates the preprocessor by coding it from scratch rather than loading in the preprocessor that was pickled in the model fitting script.
<https://github.com/UBC-MDS/DSCI522-2425-group31_age-group-prediction/commit/05f6d7436adf9dae9da6e906b863d3913c1e8464>

- [x] As mentioned I liked all the figure provided, however it wasn't clear to me what the features all are in your data. For example, the feature 'oral', I'm not certain to what this refers?

These are comments from another reviewer

**Comment by DongchunChen** [comment]
- [x] About the target of the prediction. You made the year group as the target and it makes sense because given the limited age information, we can predict age groups to enable more optimal allocation of healthcare resources. However, age information is relatively easy to collect. For this dataset, there might be more interesting stories to tell. For example, you can also use diabetes classification as the prediction target. Over time, the data can be updated year by year, and you can predict who will transit from "no diabetes" or "borderline" to "new diabetes." With these dynamic predictions, healthcare resources can be better allocated. If you want to use this project to showcase to recruiters in the future, I think this story could be worth considering.
Appreciate the comment but this is not we are trying to focus on.

- [x] About metric statement. In the dataset, the "senior" group is a minority, but I believe you are more focused on predicting the "senior" group. Therefore, it would be better to highlight that the focus of the prediction is on the "senior" group, with particular attention to its F1 score.

- [x] The input of "07_model_evaluation.py". Another reviewer also mentioned that you should call the model from "06_model_fitting.py" rather than transforming and fitting again in evaluation. In addition, the confusion matrix by "07_model_evaluation.py"(346 133 32 59) is not aligned to that by notebook "age_prediction_report.ipynb" (371 108 40 51).
<https://github.com/UBC-MDS/DSCI522-2425-group31_age-group-prediction/commit/05f6d7436adf9dae9da6e906b863d3913c1e8464> scripts modified in the commit 
we also remove the notebook folder only keep the reports folder in <https://github.com/UBC-MDS/DSCI522-2425-group31_age-group-prediction/commit/7b57b99f3ba011a3d2df35b1d7b7f6f360a08dfc>

- [x] About reference. Four references(papers) have DOIs, and I have collected them. You can add the DOIs in your references if you want. ***Updated as below and included in the references.bib. However, Quarto does not render the DOI information in the References for those that have that field.***
https://doi.org/10.1093/bioinformatics/bth457 - This is not our article, and I am not sure which reference the reviewer thought this matches in our bibliography
https://doi.org/10.1038/s41586-020-2649-2 - Confirmed and included in bibliography
https://doi.org/10.25080/Majora-92bf1922-00a - Confirmed and included in bibliography
https://doi.org/10.3102/1076998619832248 - This is not our article. We do not have a DOI for "Scikit-learn: Machine learning in python journal of machine learning research"

**Comment by DongchunChen** [comment]
- [x] One suggestion is to provide a direct link to include functionality documentation. This would make it more accessible to users. Clearly documentation of the functionality would help users to execute your code faster.

- [x] One area your team could have done is to clarify the methods used in the project. I would recommend write a 50 words intro including what specific methods that have been used and the purpose of the project to inform the user more.
E.g. This project utilizes Linear Regression models to predict housing prices based on various key features of the properties. Our methods include thorough data preprocessing, feature selection, and normalization to ensure that the model performs optimally. We conduct exploratory data analysis (EDA) to identify trends, relationships, and potential outliers in the data. Our approach also incorporates hyperparameter tuning to achieve the best possible model performance. We evaluate the model's effectiveness using metrics such as Mean Absolute Error (MAE), Mean Squared Error (MSE), and Root Mean Squared Error (RMSE). Furthermore, cross-validation is employed to ensure the model's robustness and generalizability to unseen data.
Already done in each of the sections

**Comment by Rashidikn1** [comment]
- [x] Summary in README should include high-level interpretation of analysis findings, as well 1-2 sentences on what they might mean (at least at a high-level).

**Comment by Rashidikn1** [comment]
ALL DONE, GOOD JOB GUYS!
